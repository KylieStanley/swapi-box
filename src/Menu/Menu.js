import React from 'react';
import Button from '../Button/Button'
import './Menu.scss'


const Menu = (props) => {

  return (
    <div class="nav">
      <input class="menu-toggler" id="menu-toggler" type="checkbox"></input>
      <label for="menu-toggler">$</label>
      <ul>
        <Button name={ 'Vehicles' }/>
        <Button name={ 'people' }/>
        <Button name={ 'planets' }/>
        <Button name={ 'favorites' }/>
      </ul>
    </div>
  )
}

export default Menu


