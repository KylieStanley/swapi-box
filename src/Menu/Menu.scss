@keyframes button-glow {
  0% {
    box-shadow: 0 0 8px #03b9ffb0, 0 0 16px #03b9ffb0, 0 0 24px #03b9ffb0, 0 0 32px #03b9ffb0, 0 0 40px #03b9ffb0, 0 0 48px #03b9ffb0, 0 0 56px #03b9ffb0;
  }
  50% {
    box-shadow: 0 0 16px #03b9ffb0, 0 0 24px #03b9ffb0, 0 0 32px #03b9ffb0, 0 0 40px #03b9ffb0, 0 0 48px #03b9ffb0, 0 0 56px #03b9ffb0, 0 0 64px #03b9ffb0;
  }
  100% {
    box-shadow: 0 0 8px #03b9ffb0, 0 0 16px #03b9ffb0, 0 0 24px #03b9ffb0, 0 0 32px #03b9ffb0, 0 0 40px #03b9ffb0, 0 0 48px #03b9ffb0, 0 0 56px #03b9ffb0;
  }
}

.nav {
  position: relative;
}

.menu-toggler {
  background-color: blue;
  position: absolute;
  display: block;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  opacity: 0;
  margin: auto;
  width: 150px;
  height: 150px;
  z-index: 2;
  cursor: pointer;
}

.menu-toggler:hover + label, .menu-toggler:hover + label:before, .menu-toggler:hover + label:after {
  transition: .2s linear;
  box-shadow: 0 0 16px #03b9ffb0, 0 0 24px #03b9ffb0, 0 0 32px #03b9ffb0, 0 0 40px #03b9ffb0, 0 0 48px #03b9ffb0, 0 0 56px #03b9ffb0, 0 0 64px #03b9ffb0;
}

.menu-toggler:checked ~ ul .menu-item {
  opacity: 1;
}
.menu-toggler:checked ~ ul .menu-item:nth-child(1) {
  transform: rotate(160deg) translateX(-115px);
}
.menu-toggler:checked ~ ul .menu-item:nth-child(2) {
  transform: rotate(205deg) translateX(-115px);
}
.menu-toggler:checked ~ ul .menu-item:nth-child(3) {
  transform: rotate(250deg) translateX(-115px);
}
.menu-toggler:checked ~ ul .menu-item:nth-child(4) {
  transform: rotate(295deg) translateX(-115px);
}
.menu-toggler:checked ~ ul .menu-item a {
  pointer-events: auto;
}

.menu-toggler + label {
  display: flex;
  align-items: center;
  box-shadow: 0 0 8px #03b9ffb0, 0 0 16px #03b9ffb0, 0 0 24px #03b9ffb0, 0 0 32px #03b9ffb0, 0 0 40px #03b9ffb0, 0 0 48px #03b9ffb0, 0 0 56px #03b9ffb0;
  justify-content: center;
  color: #0080b5;
  font-size: 40px;
  font-family: 'Jedi';
  width: 110px;
  height: 110px;
  z-index: 1;
  border-radius: 50%;
  transition: transform 0.5s, top 0.5s;
  position: absolute;
  top: 27px;
  left: 27px;
  margin: auto;
  animation: button-glow 1.5s ease-in-out;
  animation-iteration-count: 4; 
  animation-delay: 2.5s;
}

.menu-item:nth-child(1) a {
  transform: rotate(180deg);
}
.menu-item:nth-child(2) a {
  transform: rotate(180deg);
}
.menu-item:nth-child(3) a {
  transform: rotate(180deg);
}
.menu-item:nth-child(4) a {
  transform: rotate(180deg);
}

.menu-item {
  position: absolute;
  display: block;
  top: 48px;
  left: 50px;
  width: 80px;
  height: 80px;
  opacity: 0;
  transition: 0.5s;
}
.menu-item a {
  display: block;
  width: inherit;
  height: inherit;
  line-height: 80px;
  color: rgba(255, 255, 255, 0.7);
  background: rgba(255, 255, 255, 0.2);
  border-radius: 50%;
  text-align: center;
  text-decoration: none;
  font-size: 16px;
  pointer-events: none;
  transition: 0.2s;
}
.menu-item a:hover {
  box-shadow: 0 0 0 2px rgba(255, 255, 255, 0.3);
  background: rgba(255, 255, 255, 0.3);
  font-size: 44.44444px;
}