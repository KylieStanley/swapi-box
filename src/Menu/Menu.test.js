import React from 'react'
import { shallow } from 'enzyme'
import Menu from './Menu'

describe('Menu', () => {
  it('should match the snapshot', () => {
    const mockSelection = 'People'
    const wrapper = shallow(<Menu currentSelection={mockSelection} />)

    expect(wrapper).toMatchSnapshot()
  })
})
