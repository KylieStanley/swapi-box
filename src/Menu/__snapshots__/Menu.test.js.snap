// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    currentSelection="People"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="menu-toggler"
        >
          <h3>
            $
          </h3>
          <input
            className="menu-toggler"
            id="menu-toggler"
            type="checkbox"
          />
          <ul>
            <Button
              favorites={Array []}
              name="vehicles"
            />
            <Button
              favorites={Array []}
              name="People"
            />
            <Button
              favorites={Array []}
              name="Planets"
            />
            <Button
              name="Favorites"
            />
          </ul>
        </label>,
        <div
          className="orbit orbit1"
        />,
        <div
          className="orbit orbit2"
        />,
        <div
          className="orbit orbit3"
        />,
        <div
          className="orbit orbit4"
        />,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              $
            </h3>,
            <input
              className="menu-toggler"
              id="menu-toggler"
              type="checkbox"
            />,
            <ul>
              <Button
                favorites={Array []}
                name="vehicles"
              />
              <Button
                favorites={Array []}
                name="People"
              />
              <Button
                favorites={Array []}
                name="Planets"
              />
              <Button
                name="Favorites"
              />
            </ul>,
          ],
          "htmlFor": "menu-toggler",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$",
            },
            "ref": null,
            "rendered": "$",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "menu-toggler",
              "id": "menu-toggler",
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  favorites={Array []}
                  name="vehicles"
                />,
                <Button
                  favorites={Array []}
                  name="People"
                />,
                <Button
                  favorites={Array []}
                  name="Planets"
                />,
                <Button
                  name="Favorites"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "favorites": Array [],
                  "name": "vehicles",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "favorites": Array [],
                  "name": "People",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "favorites": Array [],
                  "name": "Planets",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "favorites": undefined,
                  "name": "Favorites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "orbit orbit1",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "orbit orbit2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "orbit orbit3",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "orbit orbit4",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="menu-toggler"
          >
            <h3>
              $
            </h3>
            <input
              className="menu-toggler"
              id="menu-toggler"
              type="checkbox"
            />
            <ul>
              <Button
                favorites={Array []}
                name="vehicles"
              />
              <Button
                favorites={Array []}
                name="People"
              />
              <Button
                favorites={Array []}
                name="Planets"
              />
              <Button
                name="Favorites"
              />
            </ul>
          </label>,
          <div
            className="orbit orbit1"
          />,
          <div
            className="orbit orbit2"
          />,
          <div
            className="orbit orbit3"
          />,
          <div
            className="orbit orbit4"
          />,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                $
              </h3>,
              <input
                className="menu-toggler"
                id="menu-toggler"
                type="checkbox"
              />,
              <ul>
                <Button
                  favorites={Array []}
                  name="vehicles"
                />
                <Button
                  favorites={Array []}
                  name="People"
                />
                <Button
                  favorites={Array []}
                  name="Planets"
                />
                <Button
                  name="Favorites"
                />
              </ul>,
            ],
            "htmlFor": "menu-toggler",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$",
              },
              "ref": null,
              "rendered": "$",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "menu-toggler",
                "id": "menu-toggler",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    favorites={Array []}
                    name="vehicles"
                  />,
                  <Button
                    favorites={Array []}
                    name="People"
                  />,
                  <Button
                    favorites={Array []}
                    name="Planets"
                  />,
                  <Button
                    name="Favorites"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "favorites": Array [],
                    "name": "vehicles",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "favorites": Array [],
                    "name": "People",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "favorites": Array [],
                    "name": "Planets",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "favorites": undefined,
                    "name": "Favorites",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "orbit orbit1",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "orbit orbit2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "orbit orbit3",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "orbit orbit4",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
